src/braket/analog_hamiltonian_simulator/rydberg/validators/field_validator_util.py

```diff
@@ -104,3 +104,50 @@ def validate_net_detuning_with_warning(
                 # Return immediately if there is an atom has net detuning
                 # exceeding MAX_NET_DETUNING at a time point
                 return program
+
+
```

Comment:
Just want to remind myself? Are these four functions the only things that needs to be moved from the service side to the client side? Also it maybe good to comment on the top of each of these functions that they are only for device emulator and not for AHS local simulator [so that people won't be confused that why they are not used in this repo].

Reply:
Actually, don't we also need "validate_pattern_precision"?

Reply:
validate_pattern_precision is never used by the Device validators so I chose not to bring it over to the Default Simulator repo.

Reply:
Adding the additional comments makes sense for explaining the use of these helpers!